.ci/verify_test_api_requests.py

```diff
@@ -27,29 +27,50 @@
 def verify_test_case(cli_arguments, path):
     print("Verifying test case: '%s'." % (path))
 
-    import_module_name, arguments, expected, output_modifier = tests.api.test_api.get_api_test_info(path)
+    import_module_name, arguments, expected, is_error, output_modifier = tests.api.test_api.get_api_test_info(path)
 
     for key, value in vars(cli_arguments).items():
         if ((value is not None) or (value)):
             arguments[key] = value
 
     api_module = importlib.import_module(import_module_name)
 
-    actual = api_module.send(arguments)
+    try:
+        actual = api_module.send(arguments)
+
+        if (is_error):
+            _format_error(expected, actual, path, "Test case does not raise an error when one was expected")
+
+            return 1
+    except Exception as ex:
+        if (not is_error):
+            raise ex
+
+        expected = expected.get('message', '')
+        expected = tests.api.test_api.PYTHON_ERROR_PREFIX + expected
+        if (expected != str(ex)):
+            _format_error(expected, str(ex), path, "Test case does not raise the expected error")
+
+            return 1
+
+        return 0
 
     actual = output_modifier(actual)
 
     if (actual != expected):
-        expected_json = json.dumps(expected, indent = 4)
-        actual_json = json.dumps(actual, indent = 4)
-
-        print("ERROR: Test case does not have expected content: '%s'." % (path))
-        print(tests.api.test_api.FORMAT_STR % (expected_json, actual_json))
+        _format_error(expected, actual, path, "Test case does not have expected content")
 
         return 1
 
     return 0
 
+def _format_error(expected, actual, path, message):
+    expected_json = json.dumps(expected, indent = 4)
+    actual_json = json.dumps(actual, indent = 4)
+
+    print("ERROR: %s: '%s'." % (message, path))
+    print(tests.api.test_api.FORMAT_STR % (expected_json, actual_json))
```

Comment:
I plan to rework the error messages to make them look better and I will no longer need the function. But, we only print when test cases fail, so we need the information to understand how to fix the code. This seems contradictory to earlier conversations about making error statements clear during testing. If you could expand on this that would be helpful.

Reply:
There are more reasons for a test to fail than not producing the expected output,
e.g., bad code or raised error.

Not printing irrelevant information does make error output more clear (since you don't have to sift through irrelevant information).